Tests/OAuth/ResourceOwner/AzureResourceOwner.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the HWIOAuthBundle package.
+ *
+ * (c) Hardware.Info <opensource@hardware.info>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace HWI\Bundle\OAuthBundle\Tests\OAuth\ResourceOwner;
+
+use HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\AzureResourceOwner;
+
+class AzureResourceOwnerTest extends GenericOAuth2ResourceOwnerTest
+{
+    protected $csrf = true;
+
+    protected $userResponse = <<<json
+{
+    "id":  "1",
+    "foo": "bar"
```

Comment:
That was a problem: I could test locally, with real data and structure. but the token (id_token value form a real service)  is structured in a way I was unable to reproduce in a test. That is why there is no assert for a username or nickname etc, in the testGetUserInformation method.
I saw that this was done in a similar manner in other tests.